test/index.js

```diff
@@ -1,223 +1,420 @@
-    request({url: 'http://localhost:8000/1'}, function(err) {
-        t.ok(!err, "response should have no error");
-        t.end();
+//helper to stay dry
+function setup(serverFile, options, onReady, beforeReady) {
+    onReady = onReady || passthru;
+    beforeReady = beforeReady || passthru;
+    handler = recluster(path.join(__dirname, 'lib', serverFile), options);
+    handler.run();
+    beforeReady();
+    handler.once('ready', function() {
+        onReady();
     });
-});
+}
+
+request = request('http://localhost:8000');
+
+describe('recluster', function() {
+
+    describe('default', function() {
+        before(function(done) {
+            setup('server.js', defaultOptions, done);
+        });
 
-runTest("async server fails if readyWhen = started", 
-        {file: "server-async.js", readyWhen: 'started'}, function(t) {
-    request({url: 'http://localhost:9000/1'}, function(err) {
-        t.ok(err, "response should have error: " + err);
-        t.end();
+        it('should have no error', function(done) {
+            requestExpectHelloWorld200(done);
+        });
     });
-});
 
+    describe('ready event', function() {
+        it('should fail if waiting for ready when online', function(done) {
+            var options = m({},
+                defaultOptions, {
+                    readyWhen: 'started'
+                });
+            setup('server.js', options, function() {
+                requestExpectError(done);
+            });
+        });
 
-runTest("async server", {file: "server-async.js"}, function(t) {
-    request({url: 'http://localhost:9000/1'}, function(err) {
-        t.ok(!err, "response should have no error");
-        t.end();
+        it('should work if waiting for ready when listening', function(done) {
+            setup('server.js', defaultOptions, function() {
+                requestExpectHelloWorld200(done);
+            }, requestExpectError);
+        });
+
+        it("should work if waiting for ready when process says he's ready", function(done) {
+            var options = m({},
+                defaultOptions, {
+                    readyWhen: 'ready'
+                });
+            setup('server-manual-ready.js', options, function() {
+                requestExpectHelloWorld200(done);
+            }, requestExpectError);
+        });
     });
-});
 
-runTest("manual ready signal",
-        {file: "server-manual-ready.js", readyWhen: 'ready'}, function(t) {
-    request({url: 'http://localhost:9001/1'}, function(err) {
-        t.ok(!err, "response should have no error");
-        t.end();
+    describe('gracefully handle broken script', function() {
+        it('it should throw an error', function() {
+            handler = recluster(path.join(__dirname, 'lib', 'server-broken.js'), defaultOptions);
+            //fiesta, try everything, if at the end everything close it's a win :o
+            handler.run();
+            handler.reload();
+            handler.terminate();
```

Comment:
Is this the test equivalent to this one?

https://github.com/doxout/recluster/blob/master/test/index.js#L48-L65

Doesn't seem to me like its the same test case.